### ISSUE TITLE
Decoding problem for the wmt_ende_tokens_32k task

### ISSUE THREAD

**Issue by tobyyouup** [tensorflow/tensor2tensor#44]
Hi,
I run the basic wmt_ende_tokens_32k  problem as showed in the Walkthrough in README.md, I use 8 gpus and the loss figure is as follows:
![image](https://user-images.githubusercontent.com/7878430/27513842-0e6bff54-59a9-11e7-802a-c9b9bea5c195.png)

But when I run decoding with  decode_from_file, the decoding file is the standard WMT2014 en-de test set(newstest2014.en), but the decoding output means nothing, just show irrelevant decoding sentences as follows:
![image](https://user-images.githubusercontent.com/7878430/27513861-9de9dafc-59a9-11e7-97f6-d9c5c00fa14b.png)

The input and output can not match, and the BLEU score is 0.  

When I run eval script following the answer provided by  @lukaszkaiser  in [https://github.com/tensorflow/tensor2tensor/issues/36](https://github.com/tensorflow/tensor2tensor/issues/36), the bleu score is nearly 0.006.

So what's the problems? Thanks!

**Comment by lukaszkaiser** [comment]
That's very strange -- the eval BLEU should be 0.2 or more with this loss, and the accuracy should be high too. Can you rerun eval and paste your metrics graph from tensorboard? In any case, this looks like a bug, I'm trying to reproduce.

**Comment by tobyyouup** [comment]
@lukaszkaiser   Thanks，I will rerun it.  By the away, can you help to have a look at this issues？ https://github.com/tensorflow/tensor2tensor/issues/45

**Comment by zxw866** [comment]
I have the same problem. @lukaszkaiser

**Comment by zxw866** [comment]
I'm also trying to reproduce "PROBLEM = wmt_ende_tokens_32k" as showed in the Walkthrough. Many of my decoding output and input sentences are very close (the outputs look like English). And the BLEU score is close to 0. Can you help me figure out the problem? Thanks!

**Comment by lukaszkaiser** [comment]
This is very strange. What are your GPUs? Are you running 1-gpu or 8-gpu, what batch size? Maybe try `transformer_base_single_gpu` or lower learning rate if you're lowering batch size?

**Comment by zxw866** [comment]
I regenerated the data and the translation was working. Thanks!

**Comment by tobyyouup** [comment]
@lukaszkaiser   I regenerated the data and the results seem ok now, How strange it is. The training is still running, I will bring you feedback when it finishes training.

**Comment by lukaszkaiser** [comment]
Thanks! I'll close this for now, but please reopen if you see trouble!

**Comment by tobyyouup** [comment]
Hi,  @lukaszkaiser   After I regenerated the data and the results seem ok. But I have trained on 6 GPUs with 120k iterations but the BLEU score for the WMT en-de newstest2014 is still nearly 15. My experimens follow strictly to the WALKTHROUGH examples and just use the WMT en-de newstest2014 for decoding and testing BLEU.
Is it normally for this BLEU score?

The training loss curves is as below:
![image](https://user-images.githubusercontent.com/7878430/27722607-47e8b970-5d9b-11e7-8df2-c74b2c475d81.png)

And this is the eval curve:
![image](https://user-images.githubusercontent.com/7878430/27722634-63c910a4-5d9b-11e7-9746-c10e5bb8c69f.png)

The BLEU score on the WMT en-de newstest2014 is(I just decoding 4 times during the training process):
Step 22189: BLEU4 = 14.61, 46.2/21.5/11.3/6.1 (BP=0.903, ratio=1.102, sys_len=53819, ref_len=59297) Step 52676: BLEU4 = 14.82, 46.5/22.2/12.0/6.7 (BP=0.874, ratio=1.135, sys_len=52254, ref_len=59297) Step 72552: BLEU4 = 15.60, 47.1/23.1/12.7/7.2 (BP=0.879, ratio=1.129, sys_len=52527, ref_len=59297) Step 127547: BLEU4 = 15.62, 47.6/23.3/12.8/7.1 (BP=0.876, ratio=1.132, sys_len=52362, ref_len=59297)

The BELU score at step 127547 is just 15.62,but the result in the paper is higher than 25.

So is it normal?

**Comment by lukaszkaiser** [comment]
Did you try averaging the checkpoints and measuring BLEU with `get_ende_bleu` as discussed in #61 ? How do you calculate your BLEU4, how do you tokenize?
